### PR TITLE
DXCDT-396: Standardize error messages

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

This pull request aims to enhance the consistency and clarity of error messages throughout the Auth0 CLI. By standardizing error messages, we improve the user experience, making it easier for developers to troubleshoot and understand issues during CLI interactions.

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ linters:
     - whitespace
     - goimports
     - gosimple
-#    - errcheck
+    - errcheck
```
Comment (sergiught) [inline .golangci.yml:19]: Took the opportunity to update the golangci lint config as well to help with err checks, error messages and as well fix comments.

File: internal/cli/log_streams.go [hunk 2]
```diff
@@ -296,7 +297,7 @@ func (c *cli) logStreamPickerOptionsByType(desiredType logStreamType) pickerOpti
 		}
 		if len(options) == 0 {
 			return nil, fmt.Errorf(
-				"There are currently no log streams of type: %q, use 'auth0 logs streams create %s' to create one.",
+				"there are currently no log streams of type: %q, use 'auth0 logs streams create %s' to create one",
```
Comment (Widcket) [inline internal/cli/log_streams.go:300]: Should we use backticks around the command in these errors, like we're doing in other places?

File: internal/cli/roles_permissions.go [hunk 3]
```diff
@@ -149,9 +147,9 @@ func addRolePermissionsCmd(cli *cli) *cobra.Command {
 			}
 
 			var rs *management.ResourceServer
-			rs, err := cli.api.ResourceServer.Read(cmd.Context(), url.PathEscape(inputs.APIIdentifier))
```
Comment (Widcket) [inline internal/cli/roles_permissions.go:152]: Same as above.

File: internal/cli/roles_permissions.go [hunk 4]
```diff
@@ -214,9 +212,9 @@ func removeRolePermissionsCmd(cli *cli) *cobra.Command {
...
```
Comment (Widcket) [inline internal/cli/roles_permissions.go:217]: Same here.

File: internal/cli/root.go [hunk 5]
```diff
@@ -55,24 +56,23 @@ func Execute() {
 		}
 	}()
 
-	// platform specific terminal initialization:
+	// Platform specific terminal initialization:
 	// this should run for all commands,
-	// for most of the architectures there's no requirements:
+	// for most of the architectures there's no requirements:.
```
Comment (Widcket) [inline internal/cli/root.go:61]:
```suggestion
	// for most of the architectures there's no requirements.
```

File: internal/cli/root.go [hunk 6]
```diff
@@ -149,7 +149,7 @@ func addPersistentFlags(rootCmd *cobra.Command, cli *cli) {
 func addSubCommands(rootCmd *cobra.Command, cli *cli) {
 	// The order of the commands here matters.
 	// Add new commands in a place that reflect its
-	// relevance or relation with other commands:
+	// relevance or relation with other commands:.
```
Comment (Widcket) [inline internal/cli/root.go:152]:
```suggestion
	// relevance or relation with other commands.
```

File: internal/cli/root.go [hunk 7]
```diff
@@ -170,7 +170,7 @@ func addSubCommands(rootCmd *cobra.Command, cli *cli) {
 	rootCmd.AddCommand(apiCmd(cli))
 	rootCmd.AddCommand(terraformCmd(cli))
 
-	// keep completion at the bottom:
+	// Keep completion at the bottom:.
```
Comment (Widcket) [inline internal/cli/root.go:173]:
```suggestion
	// Keep completion at the bottom.
```

File: internal/cli/terraform_test.go [hunk 8]
```diff
@@ -174,7 +174,7 @@ func setupTestDIRAndImportData(t *testing.T) (string, importDataList) {
 			ImportID:     "actionID_1",
 		},
 		{
-			ResourceName: "auth0_action.MyTestAction", //NOTE: duplicate name
+			ResourceName: "auth0_action.MyTestAction", //NOTE: duplicate name.
```
Comment (Widcket) [inline internal/cli/terraform_test.go:177]:
```suggestion
			ResourceName: "auth0_action.MyTestAction", // NOTE: duplicate name.
```

**Review by willvedd — APPROVED** [review]
Possibly some nit-picky things to fuss over but overall looks like a solid n+1 (or n+86).

File: internal/cli/apis.go [hunk 9]
```diff
@@ -449,13 +449,13 @@ func deleteAPICmd(cli *cli) *cobra.Command {
 				var errs []error
 				for _, id := range ids {
 					if id != "" {
-						if _, err := cli.api.ResourceServer.Read(cmd.Context(), url.PathEscape(id)); err != nil {
-							errs = append(errs, fmt.Errorf("Unable to delete API (%s): %w", id, err))
+						if _, err := cli.api.ResourceServer.Read(cmd.Context(), id); err != nil {
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/apis.go:452]: This is already done within the Go SDK: https://github.com/auth0/go-auth0/blob/a3d8aa98408d89c10f9be24002cf217c5f2f7e32/management/management_request.go#L29

File: internal/cli/apis.go [hunk 10]
```diff
@@ -496,21 +496,24 @@ func openAPICmd(cli *cli) *cobra.Command {
 			// So here if the value is not a URL, we then check if has the length of an ID
 			// If the length check fails, we know it's a non-URL audience value
 			// This will fail for non-URL audience values with the same length as the ID
-			// But it should cover the vast majority of users
+			// But it should cover the vast majority of users.
 			if _, err := url.ParseRequestURI(inputs.ID); err == nil || len(inputs.ID) != 24 {
 				if err := ansi.Waiting(func() error {
-					api, err := cli.api.ResourceServer.Read(cmd.Context(), url.PathEscape(inputs.ID))
+					api, err := cli.api.ResourceServer.Read(cmd.Context(), inputs.ID)
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/apis.go:502]: This is already done within the Go SDK: https://github.com/auth0/go-auth0/blob/a3d8aa98408d89c10f9be24002cf217c5f2f7e32/management/management_request.go#L29

File: internal/cli/apis.go [hunk 11]
```diff
@@ -449,13 +449,13 @@ func deleteAPICmd(cli *cli) *cobra.Command {
...
-						if _, err := cli.api.ResourceServer.Read(cmd.Context(), url.PathEscape(id)); err != nil {
-							errs = append(errs, fmt.Errorf("Unable to delete API (%s): %w", id, err))
+						if _, err := cli.api.ResourceServer.Read(cmd.Context(), id); err != nil {
+							errs = append(errs, fmt.Errorf("failed to delete API with ID %q: %w", id, err))
 							continue
 						}
 
-						if err := cli.api.ResourceServer.Delete(cmd.Context(), url.PathEscape(id)); err != nil {
-							errs = append(errs, fmt.Errorf("Unable to delete API (%s): %w", id, err))
+						if err := cli.api.ResourceServer.Delete(cmd.Context(), id); err != nil {
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/apis.go:457]: This is already done within the Go SDK: https://github.com/auth0/go-auth0/blob/a3d8aa98408d89c10f9be24002cf217c5f2f7e32/management/management_request.go#L29

File: internal/cli/roles_permissions.go (hunk 3 above)
Comment (sergiught) [reply internal/cli/roles_permissions.go:152]: This is already done within the Go SDK: https://github.com/auth0/go-auth0/blob/a3d8aa98408d89c10f9be24002cf217c5f2f7e32/management/management_request.go#L29

File: internal/cli/roles_permissions.go (hunk 4 above)
Comment (sergiught) [reply internal/cli/roles_permissions.go:217]: This is already done within the Go SDK: https://github.com/auth0/go-auth0/blob/a3d8aa98408d89c10f9be24002cf217c5f2f7e32/management/management_request.go#L29

**Pull request merged**